packages/open-next/src/core/routing/middleware.ts

```diff
@@ -28,6 +28,11 @@ type MiddlewareOutputEvent = InternalEvent & {
   externalRewrite?: boolean;
 };
 
+type MiddlewareLoader = (path: string) => Promise<any>;
```

Comment:
Properly typing the return would be great

Reply:
Middleware returns a [`NextResponse`](https://nextjs.org/docs/app/building-your-application/routing/middleware#nextresponse). What are your thoughts on tests taking a dependency on next?

Reply:
That's for end user, the middleware here just returns a `Response`.

That's one of the reason i don't like mocking this kind of stuff. It's user code, compiled a first time by Next, recompiled by us to be usable here. Properly mocking this is virtually impossible.

Reply:
> Properly mocking this is virtually impossible.

I don't know about this... I think it is reasonable to test how the middleware response should be handled.